crates/ruff_db/src/vendored.rs

```diff
@@ -0,0 +1,379 @@
+use std::cell::{RefCell, RefMut};
+use std::io::{self, Read};
+use std::ops::{Deref, DerefMut};
+use std::sync::{Mutex, MutexGuard};
+
+use itertools::Itertools;
+use once_cell::sync::Lazy;
+use zip::{read::ZipFile, ZipArchive, ZipWriter};
+
+pub use path::{VendoredPath, VendoredPathBuf};
+
+pub mod path;
+
+type Result<T> = io::Result<T>;
+
+#[derive(Debug)]
+pub struct VendoredFileSystem {
+    inner: VendoredFileSystemInner,
+}
+
+impl Default for VendoredFileSystem {
+    fn default() -> Self {
+        static DATA: Lazy<Box<[u8]>> = Lazy::new(|| {
+            let mut buffer = Vec::new();
+            let cursor = io::Cursor::new(&mut buffer);
+            {
+                let mut archive = ZipWriter::new(cursor);
+                archive.finish().unwrap();
+            }
```

Comment:
Style question: do you prefer an isolated scope like I currently have, or a call to `drop()`, for this kind of pattern? Not sure which is considered more idiomatic.

```suggestion
            let mut archive = ZipWriter::new(cursor);
            archive.finish().unwrap();
            drop(archive);
```

Reply:
I prefer the sub scopes.